src/screens/NewWallet.js

```diff
@@ -105,7 +105,7 @@ function NewWallet() {
   const pinSuccess = () => {
     // Generate addresses and load data
     LOCAL_STORE.unlock();
-    wallet.generateWallet(words, '', pin, password, history);
+    wallet.generateWallet(words, '', pin, password, navigate);
```

Comment:
I think we have a great opportunity here to refactor this and stop storing the navigation object in redux and just use it in the saga as it should be

Can you please do this, or open a KTLO so we can track this

Reply:
I fully agree! The initial steps for this are already within the scope of #527, that depends on this PR.